src/Microsoft.VisualStudio.Threading/AwaitExtensions.cs

```diff
@@ -660,5 +695,155 @@ public void GetResult()
             {
             }
         }
+
+        /// <summary>
+        /// A Task awaitable that has affinity to executing callbacks synchronously on the completing callstack.
+        /// </summary>
+        public struct ExecuteContinuationSynchronouslyAwaitable
```

Comment:
:bulb: `readonly struct` can apply to all these structures

Reply:
That's an interesting idea. Since we have many awaiters in this library and I presume all of them would benefit, I'm going to save that for a different PR and apply after some testing.